tests/kernel/testUnorderedSetByBlock.cpp

```diff
@@ -0,0 +1,420 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file testUnorderedSetByBlock.cpp
+ * @ingroup Tests
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2019/01/04
+ *
+ * Functions for testing class UnorderedSetByBlock.
+ *
+ * This file is part of the DGtal library.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <unordered_set>
+#include "DGtal/kernel/PointVector.h"
+#include "DGtal/kernel/UnorderedSetByBlock.h"
+#include "DGtalCatch.h"
+///////////////////////////////////////////////////////////////////////////////
+
+using namespace std;
+using namespace DGtal;
+
+template < typename Point >
+Point randomPoint( int S )
+{
+  Point p;
+  for ( DGtal::Dimension i = 0; i < Point::dimension; i++ )
+    p[ i ] = random() % S;
```

Comment:
Where did this `random` function come from?!!

Oh it seems to be a Linux-specific function, not in the C++ specification and thus not available with Windows. Could you replace it by `rand` or use the standard library?

BTW, could it be possible to also test negative coordinates?

Reply:
I have replaced random by rand and check also negative coords